File: app/src/main/java/com/github/tvbox/osc/base/ToolbarMenuProvider.java
package com.github.tvbox.osc.base;

import androidx.annotation.MenuRes;

public interface ToolbarMenuProvider {
    @MenuRes
    default int getMenuResId() {
        return 0;
    }

    default boolean onMenuItemClick(int itemId) {
        return false;
    }

    default String getToolbarTitle() {
        return null;
    }

    default boolean enableAppBarScroll() {
        return false;
    }
}

File: app/src/main/java/com/github/tvbox/osc/base/ToolbarMenuProvider.kt
package com.github.tvbox.osc.base

import androidx.annotation.MenuRes

interface ToolbarMenuProvider {
	@get:MenuRes
	val menuResId: Int
		get() = 0

	val toolbarTitle: String?
		get() = null

	fun onMenuItemClick(itemId: Int): Boolean {
		return false
	}

	fun enableAppBarScroll(): Boolean {
		return false
	}
}
